test/functional/tests/initialize/test_initialize_load.py

```diff
@@ -0,0 +1,239 @@
+        core = []
+        for i in range(0, core_number):
+            core.append(cache.add_core(core_devices[i]))
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != core_number:
+            TestRun.fail(f"Expected cores count: {core_number}; Actual cores count: {cores_count}.")
+
+    with TestRun.step("Configure optional parameters."):
+        cache.set_cleaning_policy(cleaning_policy)
+        if cleaning_policy == CleaningPolicy.alru:
+            alru = FlushParametersAlru()
+            alru.activity_threshold = Time(milliseconds=1000)
+            alru.flush_max_buffers = 10
+            alru.staleness_time = Time(seconds=60)
+            alru.wake_up_time = Time(seconds=5)
+            cache.set_params_alru(alru)
+        if cleaning_policy == CleaningPolicy.acp:
+            acp = FlushParametersAcp()
+            acp.flush_max_buffers = 100
+            acp.wake_up_time = Time(seconds=5)
+            cache.set_params_acp(acp)
+
+    with TestRun.step("Run FIO on cache's exported object"):
+        for i in range(0, core_number):
+            Fio().create_command() \
+                .io_engine(IoEngine.libaio) \
+                .io_depth(64) \
+                .size(Size(4, Unit.GibiByte)) \
+                .read_write(ReadWrite.randrw) \
+                .block_size(Size(64, Unit.MebiByte)) \
+                .target(f"{core[i].system_path}") \
+                .run()
+
+    with TestRun.step("Stop CAS."):
+        cache.stop()
+        if len(casadm_parser.get_caches()) != 0:
+            TestRun.fail("There are still running caches after stopping service.")
+        if len(casadm_parser.get_cores(cache.cache_id)) != 0:
+            TestRun.fail("There are still running cores after stopping service.")
+
+    with TestRun.step("Load CAS."):
+        cache = casadm.load_cache(cache_dev)
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != core_number:
+            TestRun.fail(f"Expected cores count: {core_number}; Actual cores count: {cores_count}.")
+
+    with TestRun.step("Compare cache configuration before and after load."):
+        if cache_mode != cache.get_cache_mode():
+            TestRun.LOGGER.fail("Cache mode are different.")
+        if cache_line_size != cache.get_cache_line_size():
+            TestRun.LOGGER.fail("Cache line size are different.")
+        if cleaning_policy != cache.get_cleaning_policy():
+            TestRun.LOGGER.fail("Cleaning policy are different.")
+        if cleaning_policy == CleaningPolicy.alru:
+            dupa = cache.get_flush_parameters_alru()
```

Comment:
![seal_of_approval](https://user-images.githubusercontent.com/55442368/70721984-c91ee880-1cf6-11ea-95c5-9372239545d5.jpg)

Reply:
thanks for watching